src/inference/src/ie_core.cpp

```diff
@@ -121,14 +121,29 @@ void allowNotImplemented(F&& f) {
 
 ov::AnyMap flatten_sub_properties(const std::string& device, const ov::AnyMap& properties) {
     ov::AnyMap result = properties;
-    for (auto&& property : properties) {
-        auto parsed = parseDeviceNameIntoConfig(property.first);
+    if (device.find("AUTO") != std::string::npos || device.find("MULTI:") != std::string::npos ||
```

Comment:
Why MULTI: and HETERO: instead of MULTI and HETERO?

Reply:
2 test cases will be failed if we remove the ":" from "HETERO:" or "MULTI:". 
src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp#L1082
The secondary properties of virtual devices should be flattened to get the priority device before entering into the plugin. Or we will get the below exception.

Test case 1: https://github.com/yangwang201911/openvino/blob/1d53cac435eb9e30be4e848f7434f2510ff13ef2/src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp#L1117

Test case 2:
https://github.com/yangwang201911/openvino/blob/1d53cac435eb9e30be4e848f7434f2510ff13ef2/src/tests/functional/plugin/shared/include/behavior/ov_plugin/core_integration.hpp#L1095

Error log for Test case 2. Similar error could encounter in test case 1.
![image](https://user-images.githubusercontent.com/59246673/174214081-e31d3c3a-60a5-4a27-805c-2c491cd6071e.png)